src/Datadog.Trace/Logging/CustomSerilogLogProvider.cs

```diff
@@ -5,44 +5,80 @@
 
 using System;
 using System.Linq.Expressions;
+using System.Reflection;
+using System.Runtime.CompilerServices;
 using Datadog.Trace.Logging.LogProviders;
 
 namespace Datadog.Trace.Logging
 {
     internal class CustomSerilogLogProvider : SerilogLogProvider, ILogProviderWithEnricher
     {
+        private static ConditionalWeakTable<object, object> _enricherToArrayConditionalWeakTable;
         private static Func<object, IDisposable> _pushMethod;
+        private static Func<object, IDisposable> _openContext;
 
         public CustomSerilogLogProvider()
         {
-            _pushMethod = GetPush();
-        }
+            _enricherToArrayConditionalWeakTable = new();
 
-        public IDisposable OpenContext(object enricher)
-        {
-            return _pushMethod(enricher);
+            var logEnricherType = GetLogEnricherType();
+            if (GetPushMethodInfo() != null)
```

Comment:
You should store the return value of `GetPushMethodInfo` in a field to avoid calling it twice

Reply:
Done in 35efec447b758d7fd8fd7988af26a1447c2c8756